### PR TITLE
Update cargo crates

### PR DESCRIPTION
**Pull request by TheQuantumPhysicist**
I also removed sscanf crate because it gets updated so much and I don't even understand why we need it to parse a 3-field string.

sscanf is also present in libp2p code, which will be removed later.

### CODE REVIEW

**Review by muursh — APPROVED** [review]
I'm fine with all of this and back scrapping deny 100%.

**Comment by TheQuantumPhysicist** [comment]
I removed deny. I'll create an issue to deal with the required replacement.

**Comment by TheQuantumPhysicist** [comment]
#590 makes the script specs to replace cargo deny.

**Review by stanislav-tkach — APPROVED** [review]

**Comment by stanislav-tkach** [comment]
I have [discovered](https://github.com/mintlayer/mintlayer-core/pull/592) that the new version of the env_logger crate causes issues, so I reverted it for now. I will try to investigate further the root of this issue.

**Comment by TheQuantumPhysicist** [comment]
Thank you very much, Stanislav. Cheers!

**Review by iljakuklic — APPROVED** [review]

**Comment by iljakuklic** [comment]
A replacement candidate for the duplicate crate check: #598

**Review by muursh — APPROVED** [review]

**Comment by iljakuklic** [comment]
Updated the compiler used for code coverage, the original one was crashing trying to compile `p2p`.

File: rpc/src/lib.rs [hunk 1]
```diff
@@ -100,8 +99,8 @@ impl Builder {
 
 /// The RPC subsystem
 pub struct Rpc {
-    http: Option<(SocketAddr, HttpServerHandle)>,
-    websocket: Option<(SocketAddr, WsServerHandle)>,
+    http: Option<(SocketAddr, ServerHandle)>,
+    websocket: Option<(SocketAddr, ServerHandle)>,
```
[the remark this answers is not in the document]
Comment (TheQuantumPhysicist) [reply rpc/src/lib.rs:103]: Maybe later. I'm not too excited for that right now.

**Comment by TheQuantumPhysicist** [comment]
Why isn't clippy running here in CI?!

**Pull request merged**